catroid/src/main/java/org/catrobat/catroid/content/bricks/SetLookBrick.java

```diff
@@ -182,18 +182,23 @@ public Brick clone() {
 
 	@Override
 	public List<SequenceAction> addActionToSequence(Sprite sprite, SequenceAction sequence) {
-		sequence.addAction(sprite.getActionFactory().createSetLookAction(sprite, look, wait));
+		if (this instanceof SetBackgroundBrick) {
+			Sprite backgroundSprite = getSpriteToChange();
+			sequence.addAction(sprite.getActionFactory().createSetLookAction(backgroundSprite, look, wait));
+		} else {
+			sequence.addAction(sprite.getActionFactory().createSetLookAction(sprite, look, wait));
+		}
```

Comment:
ok then back to you initial question:
 - yes I think it should be necessary to do this:

if this is a `SetBackgroundBrick`, then use `getSprite()`, which [is overridden](https://github.com/Catrobat/Catroid/blob/b5a9fc8323373031b4d561cba3d69de77d094319/catroid/src/main/java/org/catrobat/catroid/content/bricks/SetBackgroundBrick.java#L36) for `SetBackgroundBrick` and always returns the background sprite.
in all other cases, the passed `sprite` is used.

Apparantly it seems to be time to stop working for today... :smiley: